docs/assumptions-and-invariants.md

```diff
@@ -0,0 +1,6 @@
+# When to use ASSUME/ASSUMPTION and when to use INVARIANT/INVARIANTS with TLC?
+
+The `ASSUME` (or `ASSUMPTION`) construct in TLA+ is used to introduce constant-level (level 0) formulas—facts that must hold about the constants and fixed structures of a specification. These constraints are evaluated before TLC begins exploring behaviors and are not interpreted as properties of system states.
+A constant-level (level 0) formula contains no state variables, no primed variables, and no temporal operators. It expresses purely mathematical truths about constants, sets, or other static objects. Such formulas might assert, for example, that a constant N is a natural number or that a given set S is nonempty. These belong in ASSUME clauses within the module.
```

Comment:
I find the last sentence of this paragraph too peremptory: as you say above, ASSUME clauses assert facts that should hold about the constant parameters. Constant-level formulas may be used elsewhere, for example
```
THEOREM \A n, a, b, c \in Nat \ {0} : n > 2 => a^n + b^n # c^n
```
and these should not appear in ASSUME clauses (also TLC may not be able to check them). I suggest dropping the last sentence.